main/src/com/google/refine/model/Column.java

```diff
@@ -47,13 +48,25 @@
 import com.google.refine.model.recon.ReconConfig;
 import com.google.refine.util.ParsingUtilities;
 
+import io.frictionlessdata.tableschema.Field;
+import io.frictionlessdata.tableschema.TypeInferrer;
+import io.frictionlessdata.tableschema.exceptions.ConstraintsException;
+import io.frictionlessdata.tableschema.exceptions.InvalidCastException;
+
 public class Column implements Jsonizable {
     final private int       _cellIndex;
     final private String    _originalName;
     private String          _name;
     private ReconConfig     _reconConfig;
     private ReconStats      _reconStats;
     
+    // from data package metadata Field.java:
+    private String type = "";
```

Comment:
Do we want to allow any type or restrict them to an enumeration?

Reply:
@wetneb I'd rather not impose restriction within our column model for now against Types ( I know I know, folks will whine and complain, but I'm against this directly in our modeling)  Later we can bring about rules and sets (and extensions can even impose them as well).  That was the long term plan back in the day and is the better way to handle enumerations or restrictions as well as casting.

Reply:
@thadguidry concretely, what is it going to look like as a user? if I want to set a type for a column, will I see a text field where I can input any string that will represent the type? it seems sloppy to me.

Reply:
@wetneb I had thought about just starting with that for now, yes ?  And then expand or let your recon or extensions begin to change its form, say for instance to a drop down of enumerated values to pick from.  Or a changing enumerated list of values depending on the context switch in a particular extension or domain need.

Reply:
@wetneb this could even be driven by an expression, no ?  Imagine providing some rules via a Python or GREL expression that controls the enumerated values available.  I feel that is very powerful and rather than having Data Types being baked in to our core, we allow for expanding domain usage.  This is reminiscent of Wikidata and Schema.org's evolving Types / Properties as well.  In fact, I am on the working group for IoT.Schema.org and there is MUCH debate on mapping between vocabularies regarding various Actions and Controls of Sensors and Systems.

Reply:
@thadguidry you are mixing up two things (see my comment below). Data types should not be free-form strings or even URIs.